include/uncertainty_planning_core/uncertainty_contact_planning.hpp

```diff
@@ -141,6 +140,133 @@ class UncertaintyPolicyExecutionResult
   }
 };
 
+#if UNCERTAINTY_PLANNING_CORE__SUPPORTED_ROS_VERSION == 2
+class ExecutionTimeLimit
+{
+public:
+  ExecutionTimeLimit(const double seconds, rclcpp::Clock::SharedPtr clock)
+    : seconds_(seconds), clock_(clock)
+  {
+    if (seconds_ < 0.0)
+    {
+      throw std::invalid_argument("Time limit cannot be less than zero seconds");
+    }
+    if (!clock_)
+    {
+      throw std::invalid_argument("No valid clock supplied");
+    }
+  }
+
+  explicit ExecutionTimeLimit(const double seconds)
+    : ExecutionTimeLimit(seconds, std::make_shared<rclcpp::Clock>())
+  {
+  }
+
+  double Seconds() const { return seconds_; }
+
+  rclcpp::Clock& Clock() { return *clock_; }
+
+private:
+  double seconds_;
```

Comment:
While not strictly necessary, let's initialize `double seconds_ = 0.0;` in both implementations

Reply:
See https://github.com/calderpg/uncertainty_planning_core/pull/3/commits/cf8ce915f230816208852918241ed1d6227c0d35. 

Not an issue here, but it's worth mentioning that copy-initialization does allow a few implicit conversions (like narrowing or promotion) whereas value-initialization (e.g. `double seconds_{};` ) or direct-list-initialization (e.g. `double seconds_{0.0};`) do not. To the best of my knowledge, that's why the latter are preferred for PODs.